README.md

```diff
@@ -13,6 +13,223 @@ for kubernetes bootstrap.
+---
+kind: Machine
+apiVersion: cluster.x-k8s.io/v1alpha2
+metadata:
+  name: my-control-plane1
+  labels:
+    cluster.x-k8s.io/cluster-name: my-cluster
+    cluster.x-k8s.io/control-plane: "true"
+    set: controlplane
+spec:
+  bootstrap:
+    configRef:
+      kind: KubeadmConfig
+      apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+      name: my-control-plane1-config
+  infrastructureRef:
+    kind: DockerMachine
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+    name: my-control-plane1-docker
+  version: "v1.14.2"
+```
+
+CAPBK main responsibility is to convert a `KubeadmConfig` bootstrap object into a cloud-init script that is
+going to turn a Machine into a Kubernetes Node using kubeadm.
+
+The cloud-init script will be saved into the `KubeadmConfig.Status.BootstrapData` and then the infrastructure provider
+(CAPD in this example) will pick up this value and proceed with the machine creation and the actual bootstrap.
+
+#### KubeadmConfig examples
+
+Bootstrap control plane node:
+```yaml
+kind: KubeadmConfig
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+metadata:
+  name: my-control-plane1-config
+spec:
+  initConfiguration:
+    nodeRegistration:
+      kubeletExtraArgs:
+        eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
+  clusterConfiguration:
+    controllerManager:
+      extraArgs:
+        enable-hostpath-provisioner: "true"
+    networking:
+      podSubnet: "10.244.0.0/16"
+      dnsDomain: ""
+      serviceSubnet: ""
+```
+
+Additional control plane nodes:
+```yaml
+kind: KubeadmConfig
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+metadata:
+  name: my-control-plane2-config
+spec:
+  joinConfiguration:
+    nodeRegistration:
+      kubeletExtraArgs:
+        eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
+    controlPlane: {}
+```
+
+worker nodes:
+```yaml
+kind: KubeadmConfig
+apiVersion: bootstrap.cluster.x-k8s.io/v1alpha2
+metadata:
+  name: my-worker1-config
+spec:
+  joinConfiguration:
+    nodeRegistration:
+      kubeletExtraArgs:
+        eviction-hard: nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%
+```
+
+### Simplifying KubeadmConfig objects
+`KubeadmConfig` object allows full control of Kubeadm init/join operations by exposing raw `InitConfiguration`,
+`ClusterConfiguration` and `JoinConfiguration` objects.
+
+> valid combinations of configuration objects are:
+> - at least one of `InitConfiguration` and `ClusterConfiguration` for the first control plane node only
+> - `JoinConfiguration` for worker nodes or additional control plane nodes
+>
+> see [examples](#kubeadmconfig-examples)
+
+By design CABPK respect all the user settings in configuration `KubeadmConfig`; in case some of those settings are missing,
+CABPK applies the following set of rules that can simplify the preparation of `KubeadmConfig`:
+
+- if `KubeadmConfig.clusterConfiguration.clusterName` is empty, CABPK will automatically set this value to
```

Comment:
What do you think about having a table for this kind of information? Something like:

CABPK will fill in some values if they are left empty with sensible defaults:

Field | Default
---|---
`KubeadmConfig.clusterConfiguration.clusterName` | `Cluster.Name`
... | ...